powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/fido2/PowerAuthCryptographyService.java

```diff
@@ -68,6 +81,25 @@ private boolean verifySignature(CollectedClientData clientDataJSON, Authenticato
         return signatureUtils.validateECDSASignature(clientDataJSONEncodedHash, signature, publicKey);
     }
 
+    private boolean checkAndPersistCounter(String applicationId, String authenticatorId, int signCount) {
+        final List<ActivationRecordEntity> activations = activationRepository.findByExternalId(applicationId, authenticatorId);
+        if (activations.size() != 1) {
+            logger.warn("Multiple activations with same external ID found, external ID: {}", authenticatorId);
```

Comment:
The message is not correct when the size is zero.

Reply:
Good point, I will update the logic.